### ISSUE TITLE
Check failed: input_array.name() != output_array (lanenet/input_tensor vs. lanenet/input_tensor)The array lanenet/input_tensor is listed in both --input_arrays and --output_arrays.

### ISSUE THREAD

**Issue by shauryad15** [MaybeShewill-CV/lanenet-lane-detection#366]
Hi,

I'm trying to convert the trained model into the tflite format. But i'm getting the Check failed: input_array.name() != output_array (lanenet/input_tensor vs. lanenet/input_tensor)The array lanenet/input_tensor is listed in both --input_arrays and --output_arrays. error. which means both input and output nodes have the same name.

How can i change the name of any node or is there any solution to this.??

@MaybeShewill-CV

**Comment by MaybeShewill-CV** [comment]
@shauryad15 Maybe you should look for help in tflite group. I have not tested to converted the model into tflife format. Sorry for this:)

**Comment by shauryad15** [comment]
no, so i'm just intrested in changing the name of the node. "lanenet/input_tensor" is defined twice. I just want to rename 1 how can i do that?

**Comment by MaybeShewill-CV** [comment]
@shauryad15 It won't be defined twice with a same name in my opinion:)

**Comment by MaybeShewill-CV** [comment]
@shauryad15 Must be something wrong with your process:)

**Comment by shauryad15** [comment]
ok, in freeze_lanenet.py we're giving "input_tensor" as input_tensor as well as output tensor.... why is it needed? because when I change the name of input_tensor, i get error as "input_tensor not defined". Is it required to give the same name as both input and output tensor?
@MaybeShewill-CV

**Comment by shauryad15** [comment]
def convert_ckpt_into_pb_file(ckpt_file_path, pb_file_path):
    """

    :param ckpt_file_path:
    :param pb_file_path:
    :return:
    """
    # construct compute graph
    with tf.variable_scope('lanenet'):
        input_tensor = tf.placeholder(dtype=tf.float32, shape=[1, 256, 512, 3], name=**'input_tensor'**)

    net = lanenet.LaneNet(phase='test', net_flag='vgg')
    binary_seg_ret, instance_seg_ret = net.inference(input_tensor=input_tensor, name='lanenet_model')

    with tf.variable_scope('lanenet/'):
        binary_seg_ret = tf.cast(binary_seg_ret, dtype=tf.float32)
        binary_seg_ret = tf.squeeze(binary_seg_ret, axis=0, name='final_binary_output')
        instance_seg_ret = tf.squeeze(instance_seg_ret, axis=0, name='final_pixel_embedding_output')

    # create a session
    saver = tf.train.Saver()

    sess_config = tf.ConfigProto()
    sess_config.gpu_options.per_process_gpu_memory_fraction = 0.85
    sess_config.gpu_options.allow_growth = False
    sess_config.gpu_options.allocator_type = 'BFC'

    sess = tf.Session(config=sess_config)

    with sess.as_default():
        saver.restore(sess, ckpt_file_path)

        converted_graph_def = tf.graph_util.convert_variables_to_constants(
            sess,
            input_graph_def=sess.graph.as_graph_def(),
            output_node_names=[
                **'lanenet/input1_tensor',**
                'lanenet/final_binary_output',
                'lanenet/final_pixel_embedding_output'
            ]
        )

        with tf.gfile.GFile(pb_file_path, "wb") as f:
            f.write(converted_graph_def.SerializeToString())

**Comment by MaybeShewill-CV** [comment]
@shauryad15 where should I feed the image data when doing inference if I did not freeze the input tensor node as one of the output node. I think you may get help from TensorFlow community:)

**Comment by shauryad15** [comment]
but why will you freeze input node as output node?

**Comment by MaybeShewill-CV** [comment]
@shauryad15 ==! Like I said it is the entry node of the network so I need to freeze it as output node

**Comment by shauryad15** [comment]
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/freeze_graph.py
this is the default script provided by tensoflow. Here, i can't see freezing of input node. Please educate me if I'm missing anything. Thanks. :)
